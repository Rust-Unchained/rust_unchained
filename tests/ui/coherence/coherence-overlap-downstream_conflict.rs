pub trait Foo<X> {}
pub trait Bar<X> {}
impl<X, T> Foo<X> for T where T: Bar<X> {}

impl Bar<()> for i32 {}

impl<X> Foo<X> for i32 {}
//~^ ERROR E0119

fn main() { }
