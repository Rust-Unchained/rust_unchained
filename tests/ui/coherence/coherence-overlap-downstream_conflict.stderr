error[E0119]: conflicting implementations of trait `Foo<()>` for type `i32`
  --> $DIR/coherence-overlap-downstream_conflict.rs:7:1
   |
LL | impl<X, T> Foo<X> for T where T: Bar<X> {}
   | --------------------------------------- first implementation here
...
LL | impl<X> Foo<X> for i32 {}
   | ^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `i32`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
